modules/dnn/src/layers/prior_box_layer.cpp

```diff
@@ -529,6 +526,12 @@ class PriorBoxLayerImpl CV_FINAL : public PriorBoxLayer
             if (!_aspectRatios.empty())
             {
                 l.getParameters()["aspect_ratio"] = _aspectRatios;
+                if (_stepX != _stepY)
+                {
+                    l.getParameters()["step"] = 0.0;
+                    l.getParameters()["step_h"] = _stepY;
+                    l.getParameters()["step_w"] = _stepX;
+                }
```

Comment:
Why this parameters under `if (!_aspectRatios.empty())` condition but not in the `else` section of `if (_stepX == _stepY)`?